src/Metadata/AggregateRoot/AggregateRootMetadata.php

```diff
@@ -17,6 +17,7 @@ public function __construct(
         public readonly bool $suppressAll,
         public readonly ?string $snapshotStore,
         public readonly ?int $snapshotBatch,
+        public readonly int|string|null $snapshotVersion = null,
```

Comment:
Why do we also allow string? Do we expect such large numbers which won't fit in int?

Reply:
I would use I would always use uuids, that's much easier to manage. You don't have to be careful that you count up correctly. But I still want to give the possibility to use a number.

Reply:
So maybe just drop `int`? You could pass it also like `'1'` if someone wants a number.